### DISCUSSION TITLE
Button value

### DISCUSSION THREAD

**Discussion by jojogib** [RomanKubiak/ctrlr#528] [Q&A] [answered]
Hey every one thank you for your help so far.
I have another issue which drives me nuts.

I have 3 buttons and I want to set specific values to my button to calculate in LUA.

0 = Off 

Button a (0 1)
Button b (0 2)
Button c (0 4)

Then i want to calculate
Button a (1) + Button b (2) = 3
Button b (2) + Button c (4) = 7
...
The problem is, it don't matter whats in the "Buttons values box".. On is always 1 and Off 0
Conclusion: it counts the rows but not the actual value whats inside
So the value box works for sysexx cc .... but not for this calculation.

```

 local a = panel:getModulatorByName("a"):getValue()
 local b = panel:getModulatorByName("b"):getValue() 
 local c = panel:getModulatorByName("c"):getValue()

 local x = a+b+c

```
Anyone knows how to set this up?

Thanks

**Comment by Ted-juh** [comment, marked as answer]
Why is b+c 7 and not 6?

You are taking the value of the button which is 0 or 1. Thus assign new values with an if/else statement if button b or button c is on or off.

If b == 1 then b = 2 end
If c == 1 then c = 4 end 

Local x = a+b+c

**Reply by jojogib** [reply]
ok thanks it worked the correct way was
If b == 1 then b = 2 end
If c == 1 then c = 4 end
and yeah its 6 of course

**Reply by Ted-juh** [reply]
Yeah, my apologies, I wrote it in a hurry.
